### PR TITLE
test: longer timeout for gke cluster creation

### PR DESCRIPTION
**Pull request by sdowell**
Cluster creation via gcloud waits for the operation but has been occasionally timing out in CI. This change uses --async with a manual wait for the operation with our own timeout.

### CODE REVIEW

File: e2e/nomostest/clusters/gke.go [hunk 1]
```diff
@@ -108,8 +112,25 @@ func waitOperation(ctx context.Context, t testing.NTB, operation string) error {
...
+type operationOption func(opts *operationOptions)
+
+func withOperationTimeout(timeout time.Duration) operationOption {
+	return func(opts *operationOptions) {
+		opts.timeout = timeout
+	}
+}
+
+func listAndWaitForOperations(ctx context.Context, t testing.NTB, name string, opts ...operationOption) error {
+	options := &operationOptions{
```
Comment (karlkfi) [inline e2e/nomostest/clusters/gke.go:127]: do we need all this operationOptions complexity? why not just pass in a context with a timeout defined by the caller?

Comment (karlkfi) [reply e2e/nomostest/clusters/gke.go:127]: If you want to define a default timeout you can use ctx.Deadline() to see if the context has a timeout or not.

Comment (sdowell) [reply e2e/nomostest/clusters/gke.go]: done

File: e2e/nomostest/clusters/gke.go [hunk 2]
```diff
@@ -109,8 +113,14 @@ func waitOperation(ctx context.Context, t testing.NTB, operation string) error {
 }
 
 func listAndWaitForOperations(ctx context.Context, t testing.NTB, name string) error {
-	ctxWithTimeout, cancel := context.WithTimeout(ctx, 5*time.Minute)
-	defer cancel()
+	var ctxWithTimeout context.Context
```
Comment (karlkfi) [inline e2e/nomostest/clusters/gke.go:116]: you could just replace the ctx var. that would avoid the else branch

Comment (sdowell) [reply e2e/nomostest/clusters/gke.go]: done

**Review by karlkfi — APPROVED** [review]
/lgtm

**Pull request merged**